crates/supervisor/core/src/syncnode/node.rs

```diff
@@ -0,0 +1,552 @@
+//! [`NodeSubscriber`] implementation for subscribing to the events from managed node.
+
+use crate::{syncnode::SubscriptionError, types::ManagedEvent};
+use alloy_primitives::U256;
+use jsonrpsee::{
+    core::{
+        client::{Subscription, SubscriptionClientT},
+        middleware::RpcServiceBuilder,
+    },
+    rpc_params,
+    ws_client::{HeaderMap, HeaderValue, WsClientBuilder},
+};
+use std::{path::PathBuf, sync::Arc};
+use tokio::{sync::watch, task::JoinHandle};
+use tracing::{debug, error, info, warn};
+
+/// Configuration for the managed node.
+#[derive(Debug)]
+pub struct ManagedNodeConfig {
+    /// The chain ID of the L2
+    pub chain_id: U256,
+    /// The URL of the managed node
+    pub url: String,
+    /// The port of the managed node
+    pub port: u16,
```

Comment:
I think we can just have a single field for the endpoint. CLI is also taking url as a single endpoint.
```suggestion
    pub endpoint: String,
```